sm6125.c

```diff
@@ -0,0 +1,278 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2022, Linaro Limited */
+
+#include <sys/mman.h>
+#include <err.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "debugcc.h"
+
+static struct debug_mux cpu_cc = {
+	.phys = 0xf111000,
+	.size = 0x1000,
+	.block_name = "cpu",
+
+	.enable_reg = 0x1c,
+	.enable_mask = BIT(0),
+
+	.mux_reg = 0x1c,
+	.mux_mask = 0x3ff << 8,
+	.mux_shift = 8,
+
+	.div_reg = 0x1c,
+	.div_mask = 0xf << 28,
+	.div_shift = 28,
+	.div_val = 8,
+};
+
+static struct debug_mux gcc = {
+	.phys = 0x1400000,
+	.size = 0x1f0000,
+
+	.enable_reg = 0x30004,
+	.enable_mask = BIT(0),
+
+	.mux_reg = 0x62000,
+	.mux_mask = 0x3ff,
+
+	.div_reg = 0x30000,
+	.div_mask = 0xf,
+	.div_val = 4,
```

Comment:
My downstream sources say that the GCC divider for Trinket is 1, not 4 (you have 8, 4, 2 correct for cpu, disp and gpucc respectively).

Also, where's video_cc?

Reply:
@MarijnS95 could you testboot (your trinket tree is in a better shape..) and check the clocks reported?

about videocc.. it' s not like venus works anyway ;)

(i'll add it)

Reply:
> your trinket tree is in a better shape..

Yes :wink: 

```
sony-pdx201 somainline # ./debugcc -p sm6125 -a
                                        perfcl_clk: 37.496800MHz (37496800Hz)
                                         pwrcl_clk: 207.594752MHz (207594752Hz)
                                          snoc_clk: 239.988792MHz (239988792Hz)
                                          bimc_clk: off
                                          pnoc_clk: 133.333064MHz (133333064Hz)
                               gcc_ahb2phy_csi_clk: off
                               gcc_ahb2phy_usb_clk: 4.687100MHz (4687100Hz)
                                    gcc_apc_vs_clk: off
                              gcc_boot_rom_ahb_clk: 74.948680MHz (74948680Hz)
                                gcc_camera_ahb_clk: off
                                 gcc_camera_xo_clk: 19.200000MHz (19200000Hz)
                             gcc_camss_cci_ahb_clk: off
                                 gcc_camss_cci_clk: off
                          gcc_camss_cphy_csid0_clk: off
                          gcc_camss_cphy_csid1_clk: off
                          gcc_camss_cphy_csid2_clk: off
                          gcc_camss_cphy_csid3_clk: off
                             gcc_camss_cpp_ahb_clk: off
                             gcc_camss_cpp_axi_clk: off
                                 gcc_camss_cpp_clk: off
                           gcc_camss_cpp_tsctr_clk: off
                        gcc_camss_cpp_vbif_ahb_clk: off
                            gcc_camss_csi0_ahb_clk: off
                                gcc_camss_csi0_clk: off
                        gcc_camss_csi0phytimer_clk: off
                             gcc_camss_csi0pix_clk: off
                             gcc_camss_csi0rdi_clk: off
                            gcc_camss_csi1_ahb_clk: off
                                gcc_camss_csi1_clk: off
                        gcc_camss_csi1phytimer_clk: off
                             gcc_camss_csi1pix_clk: off
                             gcc_camss_csi1rdi_clk: off
                            gcc_camss_csi2_ahb_clk: off
                                gcc_camss_csi2_clk: off
                        gcc_camss_csi2phytimer_clk: off
                             gcc_camss_csi2pix_clk: off
                             gcc_camss_csi2rdi_clk: off
                            gcc_camss_csi3_ahb_clk: off
                                gcc_camss_csi3_clk: off
                             gcc_camss_csi3pix_clk: off
                             gcc_camss_csi3rdi_clk: off
                            gcc_camss_csi_vfe0_clk: off
                            gcc_camss_csi_vfe1_clk: off
                             gcc_camss_csiphy0_clk: off
                             gcc_camss_csiphy1_clk: off
                             gcc_camss_csiphy2_clk: off
                             gcc_camss_csiphy3_clk: off
                                 gcc_camss_gp0_clk: off
                                 gcc_camss_gp1_clk: off
                           gcc_camss_ispif_ahb_clk: off
                            gcc_camss_jpeg_ahb_clk: off
                            gcc_camss_jpeg_axi_clk: off
                                gcc_camss_jpeg_clk: off
                          gcc_camss_jpeg_tsctr_clk: off
                               gcc_camss_mclk0_clk: off
                               gcc_camss_mclk1_clk: off
                               gcc_camss_mclk2_clk: off
                               gcc_camss_mclk3_clk: off
                           gcc_camss_micro_ahb_clk: off
                    gcc_camss_throttle_nrt_axi_clk: off
                     gcc_camss_throttle_rt_axi_clk: off
                             gcc_camss_top_ahb_clk: off
                            gcc_camss_vfe0_ahb_clk: off
                                gcc_camss_vfe0_clk: off
                         gcc_camss_vfe0_stream_clk: off
                            gcc_camss_vfe1_ahb_clk: off
                                gcc_camss_vfe1_clk: off
                         gcc_camss_vfe1_stream_clk: off
                           gcc_camss_vfe_tsctr_clk: off
                        gcc_camss_vfe_vbif_ahb_clk: off
                        gcc_camss_vfe_vbif_axi_clk: off
                                   gcc_ce1_ahb_clk: 74.900344MHz (74900344Hz)
                                   gcc_ce1_axi_clk: 133.332768MHz (133332768Hz)
                                       gcc_ce1_clk: 200.000108MHz (200000108Hz)
                     gcc_cfg_noc_usb3_prim_axi_clk: 66.666312MHz (66666312Hz)
                                 gcc_cpuss_ahb_clk: 19.200000MHz (19200000Hz)
                                gcc_cpuss_gnoc_clk: 806.396044MHz (806396044Hz)
                       gcc_cpuss_throttle_core_clk: 806.397216MHz (806397216Hz)
                         gcc_cpuss_throttle_xo_clk: 19.199704MHz (19199704Hz)
                                  gcc_disp_ahb_clk: off
                        gcc_disp_gpll0_div_clk_src: 599.998276MHz (599998276Hz)
                               gcc_disp_hf_axi_clk: 25.199384MHz (25199384Hz)
                        gcc_disp_throttle_core_clk: 25.199676MHz (25199676Hz)
                                   gcc_disp_xo_clk: 19.199704MHz (19199704Hz)
                                       gcc_gp1_clk: off
                                       gcc_gp2_clk: off
                                       gcc_gp3_clk: off
                               gcc_gpu_cfg_ahb_clk: 74.946336MHz (74946336Hz)
                             gcc_gpu_gpll0_clk_src: 600.003552MHz (600003552Hz)
                         gcc_gpu_gpll0_div_clk_src: 299.999356MHz (299999356Hz)
                            gcc_gpu_memnoc_gfx_clk: off
                          gcc_gpu_snoc_dvm_gfx_clk: off
                         gcc_gpu_throttle_core_clk: off
                           gcc_gpu_throttle_xo_clk: off
                                    gcc_mss_vs_clk: off
                                      gcc_pdm2_clk: off
                                   gcc_pdm_ahb_clk: off
                                   gcc_pdm_xo4_clk: off
                                  gcc_prng_ahb_clk: 74.945752MHz (74945752Hz)
                       gcc_qmip_camera_nrt_ahb_clk: off
                        gcc_qmip_camera_rt_ahb_clk: off
                        gcc_qmip_cpuss_cfg_ahb_clk: 74.940184MHz (74940184Hz)
                             gcc_qmip_disp_ahb_clk: off
                          gcc_qmip_gpu_cfg_ahb_clk: 74.952196MHz (74952196Hz)
                     gcc_qmip_video_vcodec_ahb_clk: off
                       gcc_qupv3_wrap0_core_2x_clk: 200.000108MHz (200000108Hz)
                          gcc_qupv3_wrap0_core_clk: 100.000272MHz (100000272Hz)
                            gcc_qupv3_wrap0_s0_clk: off
                            gcc_qupv3_wrap0_s1_clk: off
                            gcc_qupv3_wrap0_s2_clk: off
                            gcc_qupv3_wrap0_s3_clk: off
                            gcc_qupv3_wrap0_s4_clk: 7.372308MHz (7372308Hz)
                            gcc_qupv3_wrap0_s5_clk: off
                       gcc_qupv3_wrap1_core_2x_clk: 199.999228MHz (199999228Hz)
                          gcc_qupv3_wrap1_core_clk: 100.000564MHz (100000564Hz)
                            gcc_qupv3_wrap1_s0_clk: off
                            gcc_qupv3_wrap1_s1_clk: off
                            gcc_qupv3_wrap1_s2_clk: off
                            gcc_qupv3_wrap1_s3_clk: off
                            gcc_qupv3_wrap1_s4_clk: off
                            gcc_qupv3_wrap1_s5_clk: off
                        gcc_qupv3_wrap_0_m_ahb_clk: 133.334528MHz (133334528Hz)
                        gcc_qupv3_wrap_0_s_ahb_clk: 4.687100MHz (4687100Hz)
                        gcc_qupv3_wrap_1_m_ahb_clk: 133.334236MHz (133334236Hz)
                        gcc_qupv3_wrap_1_s_ahb_clk: 74.898880MHz (74898880Hz)
                                 gcc_sdcc1_ahb_clk: off
                                gcc_sdcc1_apps_clk: off
                            gcc_sdcc1_ice_core_clk: 19.199412MHz (19199412Hz)
                                 gcc_sdcc2_ahb_clk: off
                                gcc_sdcc2_apps_clk: off
                         gcc_sys_noc_cpuss_ahb_clk: 19.200000MHz (19200000Hz)
                       gcc_sys_noc_ufs_phy_axi_clk: 199.999816MHz (199999816Hz)
                     gcc_sys_noc_usb3_prim_axi_clk: 66.667776MHz (66667776Hz)
                               gcc_ufs_phy_ahb_clk: 74.891264MHz (74891264Hz)
                               gcc_ufs_phy_axi_clk: 199.999520MHz (199999520Hz)
                          gcc_ufs_phy_ice_core_clk: 300.002288MHz (300002288Hz)
                           gcc_ufs_phy_phy_aux_clk: 19.199704MHz (19199704Hz)
                       gcc_ufs_phy_rx_symbol_0_clk: off
                       gcc_ufs_phy_tx_symbol_0_clk: off
                       gcc_ufs_phy_unipro_core_clk: 150.000336MHz (150000336Hz)
                         gcc_usb30_prim_master_clk: 66.665724MHz (66665724Hz)
                      gcc_usb30_prim_mock_utmi_clk: 19.199704MHz (19199704Hz)
                          gcc_usb30_prim_sleep_clk: 0.034420MHz (34420Hz)
                     gcc_usb3_prim_phy_com_aux_clk: 19.199704MHz (19199704Hz)
                        gcc_usb3_prim_phy_pipe_clk: off
                                   gcc_vdda_vs_clk: off
                                  gcc_vddcx_vs_clk: off
                                  gcc_vddmx_vs_clk: off
                                 gcc_video_ahb_clk: off
                                gcc_video_axi0_clk: off
                       gcc_video_throttle_core_clk: off
                                  gcc_video_xo_clk: 19.199704MHz (19199704Hz)
                               gcc_vs_ctrl_ahb_clk: off
                                   gcc_vs_ctrl_clk: off
                                   gcc_wcss_vs_clk: off
                           measure_only_ipa_2x_clk: 479.997948MHz (479997948Hz)
                              disp_cc_mdss_ahb_clk: 19.198384MHz (19198384Hz)
                            disp_cc_mdss_byte0_clk: 126.370000MHz (126370000Hz)
                       disp_cc_mdss_byte0_intf_clk: 63.183536MHz (63183536Hz)
                           disp_cc_mdss_dp_aux_clk: off
                        disp_cc_mdss_dp_crypto_clk: off
                          disp_cc_mdss_dp_link_clk: off
                     disp_cc_mdss_dp_link_intf_clk: off
                         disp_cc_mdss_dp_pixel_clk: off
                             disp_cc_mdss_esc0_clk: 19.198384MHz (19198384Hz)
                              disp_cc_mdss_mdp_clk: 191.998528MHz (191998528Hz)
                          disp_cc_mdss_mdp_lut_clk: 191.998528MHz (191998528Hz)
                     disp_cc_mdss_non_gdsc_ahb_clk: 19.198384MHz (19198384Hz)
                            disp_cc_mdss_pclk0_clk: 168.493136MHz (168493136Hz)
                              disp_cc_mdss_rot_clk: 19.198384MHz (19198384Hz)
                            disp_cc_mdss_vsync_clk: 19.199552MHz (19199552Hz)
                                 disp_cc_sleep_clk: 0.033392MHz (33392Hz)
                                    disp_cc_xo_clk: 19.198384MHz (19198384Hz)
                                    gpu_cc_ahb_clk: off
                                gpu_cc_crc_ahb_clk: 74.946480MHz (74946480Hz)
                                 gpu_cc_cx_apb_clk: off
                               gpu_cc_cx_gfx3d_clk: off
                           gpu_cc_cx_gfx3d_slv_clk: off
                                 gpu_cc_cx_gmu_clk: off
                            gpu_cc_cx_snoc_dvm_clk: off
                                gpu_cc_cxo_aon_clk: 19.198680MHz (19198680Hz)
                                    gpu_cc_cxo_clk: off
                                 gpu_cc_gx_cxo_clk: off
                               gpu_cc_gx_gfx3d_clk: off
                                  gpu_cc_sleep_clk: off
```

Seems sensible but not always matching `clk_summary`:

```
sony-pdx201 somainline # cat /d/clk/clk_summary
                                 enable  prepare  protect                                duty  hardware
   clock                          count    count    count        rate   accuracy phase  cycle    enable
-------------------------------------------------------------------------------------------------------
 disp_cc_xo_clk                       1        1        0           0          0     0  50000         Y
 qcom_cpufreq1                        0        0        0  2995200000          0     0  50000         Y
 qcom_cpufreq0                        0        0        0  2995200000          0     0  50000         Y
 gcc_ufs_mem_clkref_clk               0        0        0           0          0     0  50000         Y
 gcc_bimc_gpu_axi_clk                 0        0        0           0          0     0  50000         N
 gcc_sys_noc_compute_sf_axi_clk       1        1        0           0          0     0  50000         Y
 gcc_usb3_prim_clkref_clk             1        1        0           0          0     0  50000         Y
 gcc_vs_ctrl_ahb_clk                  0        0        0           0          0     0  50000         N
 gcc_video_xo_clk                     1        1        0           0          0     0  50000         Y
 gcc_video_throttle_core_clk          0        0        0           0          0     0  50000         N
 gcc_video_axi0_clk                   0        0        0           0          0     0  50000         N
 gcc_video_ahb_clk                    1        1        0           0          0     0  50000         Y
 gcc_usb3_prim_phy_pipe_clk           0        0        0           0          0     0  50000         Y
 gcc_usb30_prim_sleep_clk             1        1        0           0          0     0  50000         Y
 gcc_ufs_phy_tx_symbol_0_clk          0        0        0           0          0     0  50000         Y
 gcc_ufs_phy_rx_symbol_0_clk          0        0        0           0          0     0  50000         Y
 gcc_ufs_phy_ahb_clk                  0        0        0           0          0     0  50000         Y
 gcc_sys_noc_cpuss_ahb_clk            1        1        0           0          0     0  50000         Y
 gcc_sdcc2_ahb_clk                    0        0        0           0          0     0  50000         N
 gcc_sdcc1_ahb_clk                    0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap_1_s_ahb_clk           0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap_1_m_ahb_clk           0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap_0_s_ahb_clk           0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap_0_m_ahb_clk           0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap1_core_clk             0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap1_core_2x_clk          0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap0_core_clk             0        0        0           0          0     0  50000         N
 gcc_qupv3_wrap0_core_2x_clk          0        0        0           0          0     0  50000         N
 gcc_qmip_video_vcodec_ahb_clk        0        0        0           0          0     0  50000         N
 gcc_qmip_gpu_cfg_ahb_clk             0        0        0           0          0     0  50000         N
 gcc_qmip_disp_ahb_clk                0        0        0           0          0     0  50000         N
 gcc_qmip_camera_rt_ahb_clk           0        0        0           0          0     0  50000         N
 gcc_qmip_camera_nrt_ahb_clk          0        0        0           0          0     0  50000         N
 gcc_prng_ahb_clk                     0        0        0           0          0     0  50000         N
 gcc_pdm_xo4_clk                      0        0        0           0          0     0  50000         N
 gcc_pdm_ahb_clk                      0        0        0           0          0     0  50000         N
 gcc_gpu_throttle_xo_clk              0        0        0           0          0     0  50000         N
 gcc_gpu_throttle_core_clk            0        0        0           0          0     0  50000         N
 gcc_gpu_snoc_dvm_gfx_clk             0        0        0           0          0     0  50000         N
 gcc_gpu_memnoc_gfx_clk               0        0        0           0          0     0  50000         Y
 gcc_gpu_cfg_ahb_clk                  1        1        0           0          0     0  50000         Y
 gcc_disp_xo_clk                      1        1        0           0          0     0  50000         Y
 gcc_disp_throttle_core_clk           0        0        0           0          0     0  50000         Y
 gcc_disp_hf_axi_clk                  2        2        0           0          0     0  50000         Y
 gcc_disp_ahb_clk                     2        2        0           0          0     0  50000         Y
 gcc_cpuss_gnoc_clk                   1        1        0           0          0     0  50000         Y
 gcc_ce1_clk                          0        0        0           0          0     0  50000         Y
 gcc_ce1_axi_clk                      0        0        0           0          0     0  50000         Y
 gcc_ce1_ahb_clk                      0        0        0           0          0     0  50000         Y
 gcc_camss_vfe_vbif_axi_clk           0        0        0           0          0     0  50000         N
 gcc_camss_vfe_tsctr_clk              0        0        0           0          0     0  50000         Y
 gcc_camss_throttle_rt_axi_clk        0        0        0           0          0     0  50000         N
 gcc_camss_throttle_nrt_axi_clk       0        0        0           0          0     0  50000         N
 gcc_camss_jpeg_axi_clk               0        0        0           0          0     0  50000         N
 gcc_camss_cpp_axi_clk                0        0        0           0          0     0  50000         N
 gcc_camera_xo_clk                    1        1        0           0          0     0  50000         Y
 gcc_camera_ahb_clk                   1        1        0           0          0     0  50000         Y
 gcc_boot_rom_ahb_clk                 0        0        0           0          0     0  50000         Y
 gcc_ahb2phy_usb_clk                  1        1        0           0          0     0  50000         Y
 gcc_ahb2phy_csi_clk                  0        0        0           0          0     0  50000         N
 sleep_clk                            0        0        0       32000          0     0  50000         Y
 xo-board                             1        1        0    19200000          0     0  50000         Y
    snoc_lpass_a_clk                  0        0        0  2147483647          0     0  50000         Y
    snoc_lpass_clk                    0        0        0  2147483647          0     0  50000         Y
    snoc_periph_a_clk                 0        0        0  2147483647          0     0  50000         Y
    snoc_periph_clk                   0        0        0  2147483647          0     0  50000         Y
    mmnrt_a_clk                       0        0        0  2147483647          0     0  50000         Y
    mmnrt_clk                         0        0        0  2147483647          0     0  50000         Y
    mmrt_a_clk                        0        0        0  2147483647          0     0  50000         Y
    mmrt_clk                          0        0        0  2147483647          0     0  50000         Y
    qup_a_clk                         0        0        0  2147483647          0     0  50000         Y
    qup_clk                           0        0        0  2147483647          0     0  50000         Y
    ln_bb_clk3_a                      0        0        0    19200000          0     0  50000         Y
    ln_bb_clk3                        0        0        0    19200000          0     0  50000         Y
    ln_bb_clk2_a                      0        0        0    19200000          0     0  50000         Y
    ln_bb_clk2                        0        0        0    19200000          0     0  50000         Y
    ln_bb_clk1_a                      0        0        0    19200000          0     0  50000         Y
    ln_bb_clk1                        0        0        0    19200000          0     0  50000         Y
    ce1_a_clk                         0        0        0  2147483647          0     0  50000         Y
    ce1_clk                           0        0        0  2147483647          0     0  50000         Y
    ipa_a_clk                         0        0        0  2147483647          0     0  50000         Y
    ipa_clk                           0        0        0  2147483647          0     0  50000         Y
    cnoc_a_clk                        0        0        0  2147483647          0     0  50000         Y
    cnoc_clk                          0        0        0  2147483647          0     0  50000         Y
    rf_clk2_a                         0        0        0    19200000          0     0  50000         Y
    rf_clk2                           0        0        0    19200000          0     0  50000         Y
    rf_clk1_a                         0        0        0    19200000          0     0  50000         Y
    rf_clk1                           0        0        0    19200000          0     0  50000         Y
    qdss_a_clk                        0        0        0    19200000          0     0  50000         Y
    qdss_clk                          0        0        0    19200000          0     0  50000         Y
    bimc_a_clk                        0        0        0  2147483647          0     0  50000         Y
    bimc_clk                          0        0        0  2147483647          0     0  50000         Y
    snoc_a_clk                        0        0        0  2147483647          0     0  50000         Y
    snoc_clk                          0        0        0  2147483647          0     0  50000         Y
    bi_tcxo_a                         0        0        0    19200000          0     0  50000         Y
    bi_tcxo                           8        8        0    19200000          0     0  50000         Y
       dsi0vco_clk                    1        1        0  2022168000          0     0  50000         Y
          dsi0n1_postdiv_clk          2        2        0  1011084000          0     0  50000         Y
             dsi0n1_postdivby2_clk       1        1        0   505542000          0     0  50000         Y
                dsi0pll               1        1        0   168514000          0     0  50000         Y
                   disp_cc_mdss_pclk0_clk_src       1        1        0   168514000          0     0  50000         Y
                      disp_cc_mdss_pclk0_clk       1        1        0   168514000          0     0  50000         Y
             dsi0pllbyte              1        1        0   126385500          0     0  50000         Y
                disp_cc_mdss_byte0_clk_src       2        2        0   126385500          0     0  50000         Y
                   disp_cc_mdss_byte0_intf_clk       1        1        0   126385500          0     0  50000         Y
                   disp_cc_mdss_byte0_clk       1        1        0   126385500          0     0  50000         Y
       disp_cc_mdss_vsync_clk_src       1        1        0    19200000          0     0  50000         Y
          disp_cc_mdss_vsync_clk       1        1        0    19200000          0     0  50000         Y
       disp_cc_mdss_rot_clk_src       1        1        0    19200000          0     0  50000         ?
          disp_cc_mdss_rot_clk        1        1        0    19200000          0     0  50000         Y
       disp_cc_mdss_esc0_clk_src       1        1        0    19200000          0     0  50000         Y
          disp_cc_mdss_esc0_clk       1        1        0    19200000          0     0  50000         Y
       disp_cc_mdss_dp_pixel_clk_src       0        0        0    19200000          0     0  50000         N
          disp_cc_mdss_dp_pixel_clk       0        0        0    19200000          0     0  50000         N
       disp_cc_mdss_dp_link_clk_src       0        0        0    19200000          0     0  50000         N
          disp_cc_mdss_dp_link_intf_clk       0        0        0    19200000          0     0  50000         N
          disp_cc_mdss_dp_link_clk       0        0        0    19200000          0     0  50000         N
       disp_cc_mdss_dp_aux_clk_src       0        0        0    19200000          0     0  50000         N
          disp_cc_mdss_dp_aux_clk       0        0        0    19200000          0     0  50000         N
       disp_cc_mdss_ahb_clk_src       1        1        0    19200000          0     0  50000         ?
          disp_cc_mdss_non_gdsc_ahb_clk       0        0        0    19200000          0     0  50000         N
          disp_cc_mdss_ahb_clk        3        3        0    19200000          0     0  50000         Y
       disp_cc_pll0                   1        1        0   768000000          0     0  50000         Y
          disp_cc_mdss_mdp_clk_src       2        2        0   192000000          0     0  50000         ?
             disp_cc_mdss_mdp_lut_clk       1        1        0   192000000          0     0  50000         Y
             disp_cc_mdss_mdp_clk       2        2        0   192000000          0     0  50000         Y
       gcc_usb30_prim_mock_utmi_clk_src       1        1        0    19200000          0     0  50000         Y
          gcc_usb30_prim_mock_utmi_clk       1        1        0    19200000          0     0  50000         Y
       gcc_vs_ctrl_clk_src            0        0        0    19200000          0     0  50000         N
          gcc_vs_ctrl_clk             0        0        0    19200000          0     0  50000         N
       gcc_usb3_prim_phy_aux_clk_src       0        0        0    19200000          0     0  50000         Y
          gcc_usb3_prim_phy_com_aux_clk       0        0        0    19200000          0     0  50000         Y
       gcc_ufs_phy_phy_aux_clk_src       0        0        0    19200000          0     0  50000         Y
          gcc_ufs_phy_phy_aux_clk       0        0        0    19200000          0     0  50000         Y
       gcc_sdcc1_ice_core_clk_src       0        0        0    19200000          0     0  50000         Y
          gcc_sdcc1_ice_core_clk       0        0        0    19200000          0     0  50000         Y
       gcc_qupv3_wrap1_s5_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap1_s5_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap1_s4_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap1_s4_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap1_s3_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap1_s3_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap1_s2_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap1_s2_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap1_s1_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap1_s1_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap1_s0_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap1_s0_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap0_s5_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap0_s5_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap0_s3_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap0_s3_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap0_s2_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap0_s2_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap0_s1_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap0_s1_clk       0        0        0    19200000          0     0  50000         N
       gcc_qupv3_wrap0_s0_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_qupv3_wrap0_s0_clk       0        0        0    19200000          0     0  50000         N
       gcc_pdm2_clk_src               0        0        0    19200000          0     0  50000         N
          gcc_pdm2_clk                0        0        0    19200000          0     0  50000         N
       gcc_gp3_clk_src                0        0        0    19200000          0     0  50000         N
          gcc_gp3_clk                 0        0        0    19200000          0     0  50000         N
       gcc_gp2_clk_src                0        0        0    19200000          0     0  50000         N
          gcc_gp2_clk                 0        0        0    19200000          0     0  50000         N
       gcc_gp1_clk_src                0        0        0    19200000          0     0  50000         N
          gcc_gp1_clk                 0        0        0    19200000          0     0  50000         N
       gcc_camss_vfe1_clk_src         0        0        0    19200000          0     0  50000         N
          gcc_camss_vfe1_stream_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi_vfe1_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_vfe1_clk          0        0        0    19200000          0     0  50000         N
       gcc_camss_vfe0_clk_src         0        0        0    19200000          0     0  50000         N
          gcc_camss_vfe0_stream_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi_vfe0_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_vfe0_clk          0        0        0    19200000          0     0  50000         N
       gcc_camss_mclk3_clk_src        0        0        0           0          0     0  50000         N
          gcc_camss_mclk3_clk         0        0        0           0          0     0  50000         N
       gcc_camss_mclk2_clk_src        0        0        0           0          0     0  50000         N
          gcc_camss_mclk2_clk         0        0        0           0          0     0  50000         N
       gcc_camss_mclk1_clk_src        0        0        0           0          0     0  50000         N
          gcc_camss_mclk1_clk         0        0        0           0          0     0  50000         N
       gcc_camss_mclk0_clk_src        0        0        0           0          0     0  50000         N
          gcc_camss_mclk0_clk         0        0        0           0          0     0  50000         N
       gcc_camss_jpeg_clk_src         0        0        0    19200000          0     0  50000         N
          gcc_camss_jpeg_clk          0        0        0    19200000          0     0  50000         N
       gcc_camss_gp1_clk_src          0        0        0    19200000          0     0  50000         N
          gcc_camss_gp1_clk           0        0        0    19200000          0     0  50000         N
       gcc_camss_gp0_clk_src          0        0        0    19200000          0     0  50000         N
          gcc_camss_gp0_clk           0        0        0    19200000          0     0  50000         N
       gcc_camss_csiphy_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_camss_cphy_csid0_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_cphy_csid1_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_cphy_csid2_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_cphy_csid3_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csiphy0_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csiphy1_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csiphy2_clk       0        0        0    19200000          0     0  50000         N
       gcc_camss_csi3_clk_src         0        0        0    19200000          0     0  50000         N
          gcc_camss_csi3rdi_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi3pix_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi3_clk          0        0        0    19200000          0     0  50000         N
       gcc_camss_csi2phytimer_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi2phytimer_clk       0        0        0    19200000          0     0  50000         N
       gcc_camss_csi2_clk_src         0        0        0    19200000          0     0  50000         Y
          gcc_camss_csi2rdi_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi2pix_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi2_clk          0        0        0    19200000          0     0  50000         N
       gcc_camss_csi1phytimer_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi1phytimer_clk       0        0        0    19200000          0     0  50000         N
       gcc_camss_csi1_clk_src         0        0        0    19200000          0     0  50000         N
          gcc_camss_csi1rdi_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi1pix_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi1_clk          0        0        0    19200000          0     0  50000         N
       gcc_camss_csi0phytimer_clk_src       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi0phytimer_clk       0        0        0    19200000          0     0  50000         N
       gcc_camss_csi0_clk_src         0        0        0    19200000          0     0  50000         Y
          gcc_camss_csi0rdi_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi0pix_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi0_clk          0        0        0    19200000          0     0  50000         N
       gcc_camss_cpp_clk_src          0        0        0    19200000          0     0  50000         N
          gcc_camss_cpp_clk           0        0        0    19200000          0     0  50000         N
       gcc_camss_cci_clk_src          0        0        0    19200000          0     0  50000         N
          gcc_camss_cci_clk           0        0        0    19200000          0     0  50000         N
       gcc_camss_ahb_clk_src          0        0        0    19200000          0     0  50000         N
          gcc_camss_vfe_vbif_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_vfe1_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_vfe0_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_top_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_micro_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_jpeg_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_ispif_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi3_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi2_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi1_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_csi0_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_cpp_vbif_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_cpp_ahb_clk       0        0        0    19200000          0     0  50000         N
          gcc_camss_cci_ahb_clk       0        0        0    19200000          0     0  50000         N
       gpll9_out_early                0        0        0  1152000000          0     0  50000         N
          gpll9_out_main              0        0        0   576000000          0     0  50000         Y
       gpll8_out_early                0        0        0   960000000          0     0  50000         Y
          gpll8_out_main              0        0        0   480000000          0     0  50000         Y
       gpll7_out_early                0        0        0   807999999          0     0  50000         N
          gpll7_out_main              0        0        0   403999999          0     0  50000         Y
       gpll6_out_early                0        0        0   768000000          0     0  50000         N
          gpll6_out_main              0        0        0   384000000          0     0  50000         Y
             gcc_sdcc1_apps_clk_src       0        0        0   192000000          0     0  50000         N
                gcc_sdcc1_apps_clk       0        0        0   192000000          0     0  50000         N
       gpll5_out_main                 0        0        0   933000000          0     0  50000         N
       gpll4_out_main                 0        0        0   806400000          0     0  50000         Y
       gpll3_out_early                0        0        0  1065999999          0     0  50000         Y
       gpll0_out_early                1        1        0   600000000          0     0  50000         Y
          gcc_vsensor_clk_src         0        0        0   600000000          0     0  50000         N
             gcc_wcss_vs_clk          0        0        0   600000000          0     0  50000         N
             gcc_apc_vs_clk           0        0        0   600000000          0     0  50000         N
             gcc_mss_vs_clk           0        0        0   600000000          0     0  50000         N
             gcc_vdda_vs_clk          0        0        0   600000000          0     0  50000         N
             gcc_vddcx_vs_clk         0        0        0   600000000          0     0  50000         N
             gcc_vddmx_vs_clk         0        0        0   600000000          0     0  50000         N
          gcc_ufs_phy_unipro_core_clk_src       0        0        0   150000000          0     0  50000         Y
             gcc_ufs_phy_unipro_core_clk       0        0        0   150000000          0     0  50000         Y
          gcc_ufs_phy_ice_core_clk_src       0        0        0   300000000          0     0  50000         Y
             gcc_ufs_phy_ice_core_clk       0        0        0   300000000          0     0  50000         Y
          gcc_ufs_phy_axi_clk_src       0        0        0   200000000          0     0  50000         Y
             gcc_sys_noc_ufs_phy_axi_clk       0        0        0   200000000          0     0  50000         Y
             gcc_ufs_phy_axi_clk       0        0        0   200000000          0     0  50000         Y
          gcc_gpu_gpll0_clk_src       0        0        0   600000000          0     0  50000         N
          gcc_disp_gpll0_div_clk_src       0        0        0   600000000          0     0  50000         N
          gpll0_out_aux2              1        1        0   300000000          0     0  50000         Y
             gcc_sdcc2_apps_clk_src       0        0        0   100000000          0     0  50000         ?
                gcc_sdcc2_apps_clk       0        0        0   100000000          0     0  50000         N
             gcc_usb30_prim_master_clk_src       3        3        0    66666666          0     0  50000         Y
                gcc_cfg_noc_usb3_prim_axi_clk       1        1        0    66666666          0     0  50000         Y
                gcc_sys_noc_usb3_prim_axi_clk       1        1        0    66666666          0     0  50000         Y
                gcc_usb30_prim_master_clk       1        1        0    66666666          0     0  50000         Y
             gcc_qupv3_wrap0_s4_clk_src       0        0        0     7372800          0     0  50003         Y
                gcc_qupv3_wrap0_s4_clk       0        0        0     7372800          0     0  50000         Y
             gcc_gpu_gpll0_div_clk_src       0        0        0   300000000          0     0  50000         N
          gpll0_out_main              0        0        0   300000000          0     0  50000         Y
 disp_cc_mdss_dp_crypto_clk_src       0        0        0           0          0     0  50000         N
    disp_cc_mdss_dp_crypto_clk        0        0        0           0          0     0  50000         N
```

Reply:
>Seems sensible but not always matching clk_summary:


expected, Linux is not aware of everything that's going on

Reply:
The `div_val` should matter though, why was it different again?

Reply:
hm.. looks like i just had it wrong